raft/storage.go

```diff
@@ -169,6 +169,11 @@ func (ms *MemoryStorage) ApplySnapshot(snap pb.Snapshot) error {
 	defer ms.Unlock()
 
 	// TODO: return ErrSnapOutOfDate?
+	msIndex := ms.snapshot.Metadata.Index
+	snapIndex := snap.Metadata.Index
+	if msIndex > snapIndex {
```

Comment:
I had not added msIndex >= snapIndex assuming that if its the same as recent snap shot why to throw error for it. Is it ok to add msIndex >= snapIndex for this?